Spigot-Server-Patches/0212-Use-TerminalConsoleAppender-for-console-improvements.patch

```diff
@@ -0,0 +1,659 @@
++        // Paper start - Log message with colors
++        //MinecraftServer.LOGGER.info(ichatbasecomponent.toPlainText());
++        MinecraftServer.LOGGER.info(org.bukkit.craftbukkit.util.CraftChatMessage.fromComponent(ichatbasecomponent,
++                net.minecraft.server.EnumChatFormat.WHITE));
++        // Paper end
+     }
+ 
+     public boolean a(int i, String s) {
+diff --git a/src/main/java/net/minecraft/server/PlayerList.java b/src/main/java/net/minecraft/server/PlayerList.java
+index 0c328782..a316848b 100644
+--- a/src/main/java/net/minecraft/server/PlayerList.java
++++ b/src/main/java/net/minecraft/server/PlayerList.java
+@@ -76,8 +76,11 @@ public abstract class PlayerList {
+ 
+     public PlayerList(MinecraftServer minecraftserver) {
+         this.cserver = minecraftserver.server = new CraftServer(minecraftserver, this);
+-        minecraftserver.console = org.bukkit.craftbukkit.command.ColouredConsoleSender.getInstance();
+-        minecraftserver.reader.addCompleter(new org.bukkit.craftbukkit.command.ConsoleCommandCompleter(minecraftserver.server));
++        // Paper start - Use TerminalConsoleAppender for colors, command completion handled in TerminalConsole
++        minecraftserver.console = new com.destroystokyo.paper.console.TerminalConsoleCommandSender();
++        //minecraftserver.console = org.bukkit.craftbukkit.command.ColouredConsoleSender.getInstance();
++        //minecraftserver.reader.addCompleter(new org.bukkit.craftbukkit.command.ConsoleCommandCompleter(minecraftserver.server));
++        // Paper end
+         // CraftBukkit end
+ 
+         this.k = new GameProfileBanList(PlayerList.a);
+@@ -140,7 +143,7 @@ public abstract class PlayerList {
+ 
+         entityplayer.spawnIn(world);
+         entityplayer.setPosition(loc.getX(), loc.getY(), loc.getZ());
+-        entityplayer.setYawPitch(loc.getYaw(), loc.getPitch()); 
++        entityplayer.setYawPitch(loc.getYaw(), loc.getPitch());
+         // Spigot end
+ 
+         // CraftBukkit - Moved message to after join
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+index d515ef0f..5a43f835 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+@@ -127,7 +127,7 @@ import io.netty.buffer.ByteBuf;
+ import io.netty.buffer.ByteBufOutputStream;
+ import io.netty.buffer.Unpooled;
+ import io.netty.handler.codec.base64.Base64;
+-import jline.console.ConsoleReader;
++// import jline.console.ConsoleReader; // Paper - JLine update
+ import org.apache.commons.lang.StringUtils;
+ import org.bukkit.NamespacedKey;
+ import org.bukkit.craftbukkit.util.CraftNamespacedKey;
+@@ -1088,9 +1088,11 @@ public final class CraftServer implements Server {
+         return logger;
+     }
+ 
+-    public ConsoleReader getReader() {
++    // Paper start - JLine update
```

Comment:
In effort to reduce diff, let's avoid commenting out stuff that won't impact behavior